mvc/src/main/java/nextstep/mvc/controller/tobe/AnnotationHandlerMapping.java

```diff
@@ -21,10 +28,43 @@ public AnnotationHandlerMapping(final Object... basePackage) {
     }
 
     public void initialize() {
+        final Reflections reflections = new Reflections(this.basePackage);
+        final Set<Class<?>> controllers = reflections.getTypesAnnotatedWith(Controller.class);
+
+        for (Class<?> controller : controllers) {
+            final Method[] methods = controller.getMethods();
+            for (Method method : methods) {
+                hasRequestMappingAnnotation(controller, method);
+            }
+        }
         log.info("Initialized AnnotationHandlerMapping!");
     }
 
+    private void hasRequestMappingAnnotation(final Class<?> controller, final Method method) {
+        if (method.getAnnotation(RequestMapping.class) != null) {
+            final RequestMapping annotation = method.getAnnotation(RequestMapping.class);
+            final RequestMethod[] requestMethods = annotation.method();
+            initHandlerExecutions(controller, method, annotation, requestMethods);
+        }
+    }
```

Comment:
early return을 하는건 어떠할까요? 그리고 전반적으로 private 메서드들의 이름이 적절한지 고민해보면 좋을 것 같아요.

```suggestion
    private void hasRequestMappingAnnotation(final Class<?> controller, final Method method) {
        if (!method.isAnnotationPresent(RequestMapping.class) {
            return;
        }   
        final RequestMapping annotation = method.getAnnotation(RequestMapping.class);
        final RequestMethod[] requestMethods = annotation.method();
        initHandlerExecutions(controller, method, annotation, requestMethods);
    }
```